Automattic/wp-calypso
# CheckoutV2 tidying: Remove checkout v2 hook and settle sidebar styling

This PR removes all of the `shouldHaveCheckoutV2`, `hasCheckoutVersion` and `useCheckoutV2` hooks that were used to A/B test v2 of our Checkout project - pbOQVh-45k-p2#comment-5821.

You'll notice the code is largely removal of the above hooks from different styles or conditional rendering blocks resulting in us falling back to the v1 Checkout experience.

Additionally, since we are collapsing v1 and v2 variants together, this PR handles a number of styling changes to ensure that v1 inherits any benefits from v2 - though we're largely falling back to v1 styles in this PR.

Related to https://github.com/Automattic/payments-shilling/issues/2754

## Testing Instructions

* We need to test a number of products here, including
    * Dotcom plans (including 100 Year plan)
    * Domains and Professional email
    * Akismet products
    * Jetpack products
    * DIFM
    * No selected site flow (such as http://calypso.localhost:3000/checkout/jetpack/jetpack_backup_t1_yearly)
    * Onboarding flows
* We ideally see the standard v1 experience along with the v1 sidebar. This includes a features list at the top, a list of products, cost overrides, the total, and any pertinent sidebar nudges (2 year upsells, DIFM next step list, renewal nudges to add soon to expire products to cart etc)
